handlers/user/search.go

```diff
@@ -0,0 +1,22 @@
+package user
+
+import (
+	"net/http"
+
+	contacts "github.com/wshaman/contacts-stub"
+	"github.com/wshaman/course-rest/lib"
+)
+
+func Search(w http.ResponseWriter, r *http.Request) {
+	// url transfer as a parameter
+	query := r.URL.Query()
+	number := query.Get("number")
+	p := contacts.LoadPersistent()
+	// find by phone
+	res, err := p.FindByPhone(number)
+	if err != nil {
+		lib.ReturnInternalError(w, err)
+		return
+	}
```

Comment:
Could you check please do we have special not found error, so if we do, we need to send it when result from 'FindByPhone' is nil and we have no errors

Reply:
in https://github.com/wshaman/contacts-stub/blob/master/yp.go

func (y *yellowPages) FindByPhone(part string) ([]Contact, error) {
	....
	if len(contacts) == 0 {
		return nil, ErrNotFound
	}
	return contacts, nil
}

this func returns ErrNotFound when len == 0